daemon: remove deprecated force-local-policy-eval-at-source option

tklauser opened the pull request:
Following its deprecation in commit 0ded29b88ffb ("daemon: Deprecate force-local-policy-eval-at-source") for Cilium v1.13, remove it for Cilium v1.14.

Keep the FORCE_LOCAL_POLICY_EVAL_AT_SOURCE macro defined in bpf code only, as this will still be needed by upcoming changes.

Ref. #15452

zacharysarah reviewed (approved):
LGTM from a docs perspective 👍🏻 

![Screenshot 2023-04-04 at 1 42 28 PM](https://user-images.githubusercontent.com/3210446/229915900-c8fb7b07-c465-4e26-b73e-b54ca726fca0.png)

😻

aspsk commented on bpf/lib/l3.h:
```diff
@@ -81,8 +81,7 @@ l3_local_delivery(struct __ctx_buff *ctx, __u32 seclabel,
 
-#if defined(USE_BPF_PROG_FOR_INGRESS_POLICY) && \
-	!defined(FORCE_LOCAL_POLICY_EVAL_AT_SOURCE)
+#if defined(USE_BPF_PROG_FOR_INGRESS_POLICY)
```
This change will break the https://github.com/cilium/cilium/pull/22006 PR (as `bpf_redirect_peer` will bypass the `ingress` at the target `lxc`).

This is ok, however, to remove the option from the userspace, as we're setting the `FORCE_LOCAL_POLICY_EVAL_AT_SOURCE` macro inside `lib/l3.h`.

So this would be cool if you revert the `bpf/` part (in the context of this PR this is a non-functional code, as nobody will ever define the `FORCE_LOCAL_POLICY_EVAL_AT_SOURCE`)

tklauser replied:
Thanks for context. I've dropped the changes in `bpf/` and amended the commit message accordingly.

pchaigno commented on Documentation/operations/upgrade.rst:
```diff
@@ -316,7 +316,8 @@ Annotations:
 
-The ``sockops-enable`` option is removed
+* The ``sockops-enable`` option was removed.
+* The ``force-local-policy-eval-at-source`` option deprecated in version 1.13 was removed.
```
Nit: Other entries seem to stick to present or present perfect verb tenses.

tklauser committed:
daemon: remove deprecated force-local-policy-eval-at-source option

Following its deprecation in commit 0ded29b88ffb ("daemon: Deprecate
force-local-policy-eval-at-source") for Cilium v1.13, remove it for
Cilium v1.14.

Keep the FORCE_LOCAL_POLICY_EVAL_AT_SOURCE macro defined in bpf code
only, as this will still be needed by upcoming changes.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
